sdk/resourcemanager/azure-resourcemanager-monitor/src/main/java/com/azure/resourcemanager/monitor/implementation/DiagnosticSettingsImpl.java

```diff
@@ -201,7 +212,7 @@ private String getResourceIdFromSettingsId(String diagnosticSettingId) {
 
     private String getNameFromSettingsId(String diagnosticSettingId) {
         String resourceId = getResourceIdFromSettingsId(diagnosticSettingId);
-        return diagnosticSettingId
+        return ResourceUtils.encodeResourceId(diagnosticSettingId)
             .substring(resourceId.length() + DiagnosticSettingImpl.DIAGNOSTIC_SETTINGS_URI.length());
     }
```

Comment:
There may be some problem here on parse the name.

The `@PathParam("name") String name` in RestProxy is not encoded. But here `encodeResourceId(diagnosticSettingId)` may get the name encoded (along with resourceUri).
Hence the name may get encoded twice in the end.

Reply:
Got it. name shouldn't be encoded, thanks!